pkg/generate/app/app.go

```diff
@@ -138,23 +140,26 @@ func (r *SourceRef) BuildSource() (*buildapi.BuildSource, []buildapi.BuildTrigge
 			},
 		},
 	}
-	var source *buildapi.BuildSource
-	switch {
-	case r.URL != nil:
-		source = &buildapi.BuildSource{
-			Type: buildapi.BuildSourceGit,
-			Git: &buildapi.GitBuildSource{
-				URI: urlWithoutRef(*r.URL),
-				Ref: r.Ref,
-			},
-			ContextDir: r.ContextDir,
-		}
-	case len(r.DockerfileContents) != 0:
+	source := &buildapi.BuildSource{}
+
+	if len(r.DockerfileContents) != 0 {
```

Comment:
shouldn't this still be a switch since the different types are mutually exclusive?